java/simulated_ble/README.md

```diff
@@ -0,0 +1,306 @@
+
+![Dataflow between three modules](dataflow.png "Input: Simulated BLE Module; Processor: Converter Module; Output: Printer Module")
+
+
+## Understand the Code
+
+### Maven Project Structure
+
+Since IoT Edge packages are based on Maven, we need to create a typical Maven project structure, which contains a `pom.xml` file.
+
+Ths POM inherits from the `com.microsoft.azure.gateway.gateway-module-base` package which declares all the dependencies needed by a module project like the runtime binaries, the gateway configuration file as well as the execution behavior. This will save us from writing hundreds of lines of code again and again!
+
+The only essential thing we need to do in our own `pom.xml` is to re-declare the dependencies and plugins used in parent, and to specify the name of the IoT Edge configuration file. Everything else is optional to module development.
+
+```xml
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+
+  <!-- Inherit from parent -->
+  <parent>
+    <groupId>com.microsoft.azure.gateway</groupId>
+    <artifactId>gateway-module-base</artifactId>
+    <version>1.0.1</version>
+  </parent>
+  
+  <groupId>com.microsoft.azure.gateway</groupId>
+  <artifactId>ble-converter</artifactId>
+  <version>1.0</version>
+  <packaging>jar</packaging>
+
+  <!-- Set the filename of the Azure IoT Edge configuration located
+       under ./src/main/resources/gateway/ which is used in parent -->
+  <properties>
+    <gw.config.fileName>gw-config.json</gw.config.fileName>
+  </properties>
+
+  <!-- Re-declare dependencies used in parent -->
+  <dependencies>
+    <dependency>
+      <groupId>com.microsoft.azure.gateway</groupId>
+      <artifactId>gateway-java-binding</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>${dependency.runtime.group}</groupId>
+      <artifactId>${dependency.runtime.name}</artifactId>
+    </dependency>
+  </dependencies>
+
+  <!-- Re-declare plugins used in parent -->
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-dependency-plugin</artifactId>
+      </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-resources-plugin</artifactId>
+      </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-shade-plugin</artifactId>
+      </plugin>
+      <plugin>
+        <groupId>org.codehaus.mojo</groupId>
+        <artifactId>exec-maven-plugin</artifactId>
+      </plugin>
+    </plugins>
+  </build>
+</project>
+```
+
+### Basic Knowledge of Module
+
+You could simply treat a module of Azure IoT Edge as a data processor whose job is to: receive input, process it, and produce output.
```

Comment:
You can treat an Azure IoT Edge module as a data processor whose job is to: receive input, process it, and produce output.